### PR TITLE
Remove LD flag for payment_mode_configurations and automated report submissions

### PR DESCRIPTION
**Pull request by arjunaj5**
https://app.clickup.com/t/30k7g3x
https://app.clickup.com/t/30k7ewy

### CODE REVIEW

**Review by gokulk16 — APPROVED** [review]

File: src/app/core/services/payment-modes.service.ts [hunk 1]
```diff
@@ -26,14 +26,11 @@ export class PaymentModesService {
 
   checkIfPaymentModeConfigurationsIsEnabled() {
     return forkJoin({
```
Comment (yksolanki9) [inline src/app/core/services/payment-modes.service.ts:28]: forkJoin is not needed here

File: src/app/core/services/launch-darkly.service.ts [hunk 2]
```diff
@@ -68,18 +68,10 @@ export class LaunchDarklyService {
     return this.getVariation('keyboard_plugin_enabled', true);
   }
 
-  checkIfPaymentModeConfigurationsIsEnabled(): Observable<boolean> {
```
Comment (yksolanki9) [inline src/app/core/services/launch-darkly.service.ts:71]: This method is being used in many places, you need to remove it from everywhere

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/core/services/launch-darkly.service.ts (hunk 2 above)
Comment (arjunaj5) [reply src/app/core/services/launch-darkly.service.ts:71]: There is one more method in `PaymentModesService` with the same name, It is being used in other places. 
This method was used only by `PaymentModesService.checkIfPaymentModeConfigurationsIsEnabled()` which is removed in this PR

File: src/app/core/services/payment-modes.service.ts (hunk 1 above)
Comment (arjunaj5) [reply src/app/core/services/payment-modes.service.ts]: fixed it

**Review by yksolanki9 — APPROVED** [review]
Please test this with and without the PMS setting enabled before merging

**Pull request merged**